mitodl/mit-open
# Article image uploading

# What are the relevant tickets?
#123 

# Description (What does it do?)
This PR re-enables image uploading in Ckeditor, and configures our server-side HTML sanitization. Additionally, I updated Ckeditor to v40.0.0 which seems to fix #115.


# Screenshots (if appropriate):
<img width="350" alt="Screenshot 2023-10-10 at 4 41 57 PM" src="https://github.com/mitodl/mit-open/assets/9010790/a57cc239-c263-4e5b-93cb-5859ad1a2134">
<img width="350" alt="Screenshot 2023-10-10 at 4 42 03 PM" src="https://github.com/mitodl/mit-open/assets/9010790/0e65d904-3b61-4aab-9d3d-ab61ee5a0376">



# How can this be tested?
1. Get `CKEDITOR_ENVIRONMENT_ID`, `CKEDITOR_SECRET_KEY`, and `CKEDITOR_UPLOAD_URL` environment variables from Open's RC heroku instance. *We use a Ckeditor-hosted image upload service, with which these env vars are associated.*
2. Visit http://localhost:8063/api/v1/articles/. If you don't have any articles yet, create a few via the JSON form. *There's no UI for creation, just editing.*) Note their ids.
3. Visit http://localhost:8063/articles/ARTICLE_ID_HERE. Edit your article and try uploading an image or two. Including:
    - Setting an alt attribute *Click on the image after you've added it*
    - try both alignments (center, side)
4. Save your article. It should look the same.

### Notes
The mediaEmbed plugin (the other button in the "+" toolbar, for embedding videos and card previews of other sites) does not fully work yet. That'll be addressed in #124
